packages/block-editor/src/hooks/utils.js

```diff
@@ -30,8 +30,75 @@ export const cleanEmptyObject = ( object ) => {
 	return isEmpty( cleanedNestedObjects ) ? undefined : cleanedNestedObjects;
 };
 
+/**
```

Comment:
It looks like we're dropping support for arrays and their indexed notation `[0]` here. Is that intentional?

Reply:
I'm happy to implement it, but IMO it would needlessly complicate the function, and I didn't see a need for that within our existing usages. 

Unless I'm misunderstanding you, in which case I'd ask you for a unit test or an example of what exactly you mean 😉

Reply:
Nope, dropping it is indeed my preferred approach, as is dropping the dot notation as well 👍

Reply:
Thanks for double-checking 👍